packages/materialized/src/complex/table-array.control.tsx

```diff
@@ -0,0 +1,155 @@
+  RankedTester,
+  rankWith,
+  registerStartupRenderer,
+  Renderer,
+  resolveSchema,
+  schemaMatches,
+  uiTypeIs,
+  update
+} from 'jsonforms-core';
+import { connect } from 'react-redux';
+
+import Table, { TableBody, TableCell, TableHead, TableRow } from 'material-ui/Table';
+import Typography from 'material-ui/Typography';
+import Button from 'material-ui/Button';
+import Toolbar from 'material-ui/Toolbar';
+
+/**
+ * Alternative tester for an array that also checks whether the 'table'
+ * option is set.
+ * @type {RankedTester}
+ */
+export const tableArrayTester: RankedTester = rankWith(10, and(
+    uiTypeIs('Control'),
+    schemaMatches(schema =>
+        !_.isEmpty(schema)
+        && schema.type === 'array'
+        && !_.isEmpty(schema.items)
+        && !Array.isArray(schema.items) // we don't care about tuples
+        && (schema.items as JsonSchema).type === 'object'
+    ))
+);
+
+export class TableArrayControl extends Renderer<ControlProps, void> {
+
+  // TODO duplicate code
+  addNewItem(path: string) {
+    const element = {};
+    this.props.dispatch(
+      update(
+        path,
+        array => {
+          if (array === undefined || array === null) {
+            return [element];
+          }
+
+          const clone = _.clone(array);
+          clone.push(element);
+
+          return clone;
+        }
+      )
+    );
+  }
+
+  /**
+   * @inheritDoc
+   */
+  render() {
+    const { uischema, schema, path, data, visible, errors, label } = this.props;
+    const controlElement = uischema as ControlElement;
+
+    const tableClass = JsonForms.stylingRegistry.getAsClassName('array-table.table');
+    const labelClass = JsonForms.stylingRegistry.getAsClassName('array-table.label');
+    const buttonClass = JsonForms.stylingRegistry.getAsClassName('array-table.button');
+    const controlClass = [JsonForms.stylingRegistry.getAsClassName('array-table'),
+      convertToClassName(controlElement.scope.$ref)].join(' ');
+
+    const resolvedSchema = resolveSchema(schema, controlElement.scope.$ref + '/items');
+    const createControlElement = (key: string): ControlElement => ({
+      type: 'Control',
+      label: false,
+      scope: { $ref: `#/properties/${key}` }
+    });
+    const labelObject = getElementLabelObject(schema, controlElement);
+    const isValid = errors.length === 0;
+    const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+
+    return (
+      <div className={controlClass} hidden={!visible}>
+        <Toolbar>
+          <Typography type='title'>{label}</Typography>
+          <Button raised color='primary' className={buttonClass}
+            onClick={ () => this.addNewItem(path) }>
+            Add to {labelObject.text}
+          </Button>
+        </Toolbar>
+        <div className={divClassNames}>
+          {!isValid ? formatErrorMessage(errors) : ''}
+        </div>
+        <Table className={tableClass}>
+          <TableHead>
+            <TableRow>
```

Comment:
Header row is not aligned with the body rows for me

Reply:
must be fixed with #729